platform/darwin/src/MGLGeometryOfflineRegion.h

```diff
@@ -0,0 +1,80 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLFoundation.h"
+#import "MGLOfflineRegion.h"
+#import "MGLShape.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An offline region defined by a style URL, geographic coordinate bounds, and
+ range of zoom levels.
+ */
+MGL_EXPORT
+@interface MGLGeometryOfflineRegion : NSObject <MGLOfflineRegion, NSSecureCoding, NSCopying>
+
+/**
+ URL of the style whose resources are required for offline viewing.
+
+ In addition to the JSON stylesheet, different styles may require different font
+ glyphs, sprite sheets, and other resources.
+
+ The URL may be a full HTTP or HTTPS URL or a Mapbox URL indicating the style’s
+ map ID (`mapbox://styles/{user}/{style}`).
+ */
+@property (nonatomic, readonly) NSURL *styleURL;
```

Comment:
Let’s promote `styleURL` to the MGLOfflineRegion protocol, since it’s shared in common between the two conforming classes and we don’t expect to introduce a new region class that doesn’t depend on a style URL. I could go either way about promoting `minimumZoomLevel` and `maximumZoomLevel`.